### PR TITLE
Expose unread and download count of Manga in category api

### LINKED ISSUE #224: [Feature Request] Implement Unread Count on Manga level
## What feature should be added to Tachidesk?
The Api should return how many Chapters of a Manga in the library are unread.

##  Why/Project's Benefit/Existing Problem
This would enable the UIs to implement a Unread Badge. Furthermore a proper implementation would make it possible to implement the Tachiyomi Style Filter Unread/Read on the Category Level.

### PR DESCRIPTION
**Pull request by mrsOwlex**
Implements unread and download counters for Mangas in Libraries Closes #224

extended /api/v1/category/{categoryId} to return the MangaViewDataClass.
This I mapped by MangaView which is a sql-view.

Feature of it are the download_count and unread_count

As previous discussions were talking about performance I included a basic test, so we can switch around to a different solution in the future.

### CODE REVIEW

File: server/src/main/kotlin/suwayomi/tachidesk/manga/model/dataclass/MangaDataClass.kt [hunk 1]
```diff
@@ -2,7 +2,7 @@ package suwayomi.tachidesk.manga.model.dataclass
 
 /*
  * Copyright (C) Contributors to the Suwayomi project
- * 
+ *
```
Comment (AriaMoradi) [inline server/src/main/kotlin/suwayomi/tachidesk/manga/model/dataclass/MangaDataClass.kt]: hmm, fix please.

**Review by AriaMoradi — CHANGES_REQUESTED** [review]

File: server/src/main/kotlin/suwayomi/tachidesk/manga/model/view/MangaView.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import suwayomi.tachidesk.manga.model.dataclass.MangaDataClass
+import suwayomi.tachidesk.manga.model.dataclass.toGenreList
+import suwayomi.tachidesk.manga.model.table.MangaStatus.Companion
+
+/**
+ * This is a dirty hack to implement View with exposed.
+ * Exposed has actually no concept of a View, so it has been manually created by a sql Migration
+ * Selects on this view work fine, but modifying queries are neither adviced nor tested
+ * One could override the modifying functions or just don't use them
+ */
```
Comment (AriaMoradi) [inline server/src/main/kotlin/suwayomi/tachidesk/manga/model/view/MangaView.kt]: I'm not really a fan of introducing Views to the codebase specially when the ORM doesn't support them. It's a neat trick and works for this case, but the solution is likely to be reused in the future and I don't want it to spread.

Sorry, we have to make due with simple queries.

**Review by AriaMoradi — CHANGES_REQUESTED** [review]

File: server/src/main/kotlin/suwayomi/tachidesk/manga/model/view/MangaView.kt (hunk 2 above)
Comment (mrsOwlex) [reply server/src/main/kotlin/suwayomi/tachidesk/manga/model/view/MangaView.kt]: Fine. I'll create it as a query then. Shouldn't be that hard now that I have a test 😋

If we would implement filters on the API I'd recommend revisiting this solution as it is much more straightforward.

Comment (mrsOwlex) [reply server/src/main/kotlin/suwayomi/tachidesk/manga/model/view/MangaView.kt]: Implemented it now i a fully supported way. Ready for a new review

**Comment by AriaMoradi** [comment]
I may do some cleanup afterwards

**Pull request merged**